### ISSUE TITLE
`SnapshotTesting.ViewImageConfig.safeArea` is ignored

### ISSUE THREAD

**Issue by xavierLowmiller** [pointfreeco/swift-snapshot-testing#180]
## Description

When snapshotting view controllers, the property `SnapshotTesting.ViewImageConfig.safeArea` is ignored.

Consider the following test case:

```swift
func testViewControllerVerticalOffset() {
    let vc = generateViewHierarchy()

    assertSnapshot(matching: vc, as: .image(on: .iPhone8))
}
```

This produces different results on the iPhone 7 and iPhone XR simulators:

| ![ref][] | ![actual][] | ![diff][] |
| --- | --- | --- |

[ref]: https://user-images.githubusercontent.com/16212751/53238058-9a1c8500-3698-11e9-920c-63a3af19f745.png
[actual]: https://user-images.githubusercontent.com/16212751/53238057-9a1c8500-3698-11e9-8a6d-90295380a59b.png
[diff]: https://user-images.githubusercontent.com/16212751/53238056-9a1c8500-3698-11e9-8401-b79e62d1790d.png

I made a minimal project that shows the issue: 
[SnapshotTest-SafeArea.zip](https://github.com/pointfreeco/swift-snapshot-testing/files/2893602/SnapshotTest-SafeArea.zip)

If somebody can point me in the right direction, I'd love to take a stab at this and make a PR.
Unfortunately, I wouldn't know where to start...

**Comment by stephencelis** [comment]
@xavierLowmiller Thanks for the report. This is a bit of a known gotcha for UINavigationController and UITabBarController: https://github.com/pointfreeco/swift-snapshot-testing/blob/71b12527f5ac243227fccfccc3758fe9d209a233/Sources/SnapshotTesting/Common/View.swift#L592-L593

We compensate by truncating non X-based sims to be the expected height, but maybe we could also crop screen shots generated by X-based sims? Ideally we find the secret sauce to trick these common controllers into rendering properly, though!

**Comment by xavierLowmiller** [comment]
Ok, this is a good pointer. Thanks!

I'll take a look and see if I can figure something out.

**Comment by stephencelis** [comment]
@xavierLowmiller I'm going to close this since we already do what we can to compensate for Apple's behavior. If you've found a fix, though, we'd love to see it!

**Comment by xavierLowmiller** [comment]
Oh yes, sorry, I completely forgot about this.

Sadly, no fix so far...

**Comment by choulepoka** [comment]
I believe this is finally Fix on Apple's side starting with Xcode 13.3, iOS SDK 15.4
